internal/packager/custom.go

```diff
@@ -28,23 +33,53 @@ type CustomPayloadCreated interface {
 	CreatedTime() time.Time
 }
 
+// CustomPayloadAppVersion is a custom payload with a docker app version
+type CustomPayloadAppVersion interface {
+	AppVersion() string
+}
+
 type payloadV1_0 struct {
+	Version string    `json:"app-version"`
 	Created time.Time `json:"created"`
 }
 
 func (p payloadV1_0) CreatedTime() time.Time {
 	return p.Created
 }
 
+func (p payloadV1_0) AppVersion() string {
+	return p.Version
+}
+
 func newCustomPayload() (json.RawMessage, error) {
-	p := payloadV1_0{Created: time.Now().UTC()}
+	p := payloadV1_0{Created: time.Now().UTC(), Version: internal.Version}
 	j, err := json.Marshal(&p)
 	if err != nil {
 		return nil, err
 	}
 	return j, nil
 }
 
+// CheckAppVersion
+func CheckAppVersion(stderr io.Writer, bndl *bundle.Bundle) error {
+	payload, err := CustomPayload(bndl)
+	if err != nil {
+		return err
+	}
+	if payload == nil {
+		return nil
+	}
+
+	var version string
+	if versionPayload, ok := payload.(CustomPayloadAppVersion); ok {
+		version = versionPayload.AppVersion()
+	}
+	if version != internal.Version {
```

Comment:
Here we are testing the versions are different, not the App Image has been built with a prior version of docker app. Maybe it's what we want, but I think the message should be different in that case, saying that App Image and docker app versions differ.